clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -1105,6 +1105,19 @@ static ParmVarDecl *getSyclKernelHandlerArg(FunctionDecl *KernelCallerFunc) {
   return (KHArg != KernelCallerFunc->param_end()) ? *KHArg : nullptr;
 }
 
+static bool isReadOnlyAccessor(const TemplateArgument &AccessModeArg) {
+  const auto *AccessModeArgEnumType =
+      AccessModeArg.getIntegralType()->castAs<EnumType>();
+  EnumDecl *ED = AccessModeArgEnumType->getDecl();
+
+  auto IsReadOnly = llvm::find_if(ED->enumerators(), [&](EnumConstantDecl *E) {
```

Comment:
FWIW, and I don't really care enough to do more than say it, this ends up being a bit less efficient than finding `read`, then returning whether it matches, since this can't early-exit.

Reply:
Can you please elaborate? Do you mean this won't be efficient if the value of `InitVal `of read and `AccessmodeArg` doesn't match?

Reply:
Right.  The earlier version searched until it found 'read', then stopped going through the loop.  The new version goes through the loop until it finds 'read' AND `InitVal`.  If the `read` element doesn't match `InitVal` this searches the whole loop.

Reply:
Yes. I can change that if you'd like. The chance of that happening is low though....Unless the library implementer decides to change the value for some reason

Reply:
I don't have opinions either way. :-)